src/processor/fingerprint.cpp

```diff
@@ -0,0 +1,283 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+#include "processor/fingerprint.hpp"
+#include "sha256.hpp"
+#include "transformer/lowercase.hpp"
+
+namespace ddwaf {
+namespace {
+
+struct string_buffer {
+    explicit string_buffer(std::size_t length)
+        // NOLINTNEXTLINE(cppcoreguidelines-no-malloc,hicpp-no-malloc,cppcoreguidelines-pro-type-reinterpret-cast)
+        : buffer(reinterpret_cast<char *>(malloc(sizeof(char) * length))), length(length)
```

Comment:
Wouldn't this leak if `move` isn't called?

Reply:
Fixed.